mkosi

```diff
@@ -457,6 +457,34 @@ def install_arch(args, workspace, run_build_script):
 
     subprocess.run(cmdline, check=True)
 
+    subprocess.run(["arch-chroot", os.path.join(workspace, "root"), "systemctl", "enable", "systemd-networkd"], check=True)
+    subprocess.run(["arch-chroot", os.path.join(workspace, "root"), "systemctl", "enable", "systemd-resolved"], check=True)
+    os.remove(os.path.join(workspace, "root", "etc/resolv.conf"))
+    os.symlink("../usr/lib/systemd/resolv.conf", os.path.join(workspace, "root", "etc/resolv.conf"))
+
+    nsswitch_old = os.path.join(workspace, "root", "etc/nsswitch.conf")
+    nsswitch_new = os.path.join(workspace, "root", "etc/nsswitch.conf.new")
+
+    with open(nsswitch_old, "r") as old:
+        with open(nsswitch_new, "w") as new:
+            for line in old:
+                if line.startswith("hosts:"):
+                    new.write(" ".join(["resolve" if x == "dns" else x for x in line.split(" ")]))
+                else:
+                    new.write(line)
+
+    shutil.copystat(nsswitch_old, nsswitch_new)
+    os.remove(nsswitch_old)
+    shutil.move(nsswitch_new, nsswitch_old)
+
+    if args.bootable:
+        with open(os.path.join(workspace, "root", "etc/systemd/network/wired.network"), "w") as f:
+            f.write("[Match]\n")
+            f.write("Name=en*\n")
+            f.write("\n")
+            f.write("[Network]\n")
+            f.write("DHCP=yes\n")
+
     print_step("Installing ArchLinux complete.")
```

Comment:
Not sure how I feel about this patch. The goal with mkosi was actualy to change as little as possible from the actual image, instead of patching around too widely.

here's an idea: maybe we should start shipping something like wired.network by default in systemd, but ask downstream distros to split it out in separate packages? That way it would be a simple matter of installing that additional package to enable automatic networkd coverage of everything ethernet. And in the case of mkosi we'd add that to the default packet set? Would that work for ArchLinux?